llvm/test/Transforms/VectorCombine/AArch64/shuffletoidentity.ll

```diff
@@ -754,10 +720,8 @@ define void @trunc(<8 x i64> %a, <8 x i64> %b, ptr %p) {
 
 define <4 x i64> @zext_chain(<4 x i16> %x) {
```

Comment:
Why didn't `add_chain` and `zext_add_chain` get updated?

Reply:
Can you kindly verify this? I'm mainly worried about some underlying bug in the algorithm, which could result in correctness issues as opposed to missed optimizations. It seems very strange to me that a single add, but not a chain of adds, is handled. If multiple uses aren't handled, even a single add with identical operands shouldn't be reduced. If a chain of adds and chain of zexts are handled, a combination of the two should be handled automatically. At least that's how my (now abandoned) algorithm worked.

Reply:
Yeah It should be correct, if not optimal. The algorithm needs to check for extra uses to make sure that there will not introduce more instructions. This isn't true of the highest shuffle though, which does not have the same one-use check requirement.